ext/gd/gd.c

```diff
@@ -4337,6 +4337,21 @@ PHP_FUNCTION(imageresolution)
 }
 /* }}} */
 
+PHP_FUNCTION(imagecompare)
+{
+	zval *IM1, *IM2;
+	gdImagePtr im1, im2;
+
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_OBJECT_OF_CLASS(IM1, gd_image_ce)
+		Z_PARAM_OBJECT_OF_CLASS(IM2, gd_image_ce)
+	ZEND_PARSE_PARAMETERS_END();
+
+	im1 = php_gd_libgdimageptr_from_zval_p(IM1);
+	im2 = php_gd_libgdimageptr_from_zval_p(IM2);
+
+	RETURN_LONG((zend_long)gdImageCompare(im1, im2));
```

Comment:
Should this be normalized to `-1`/`0`/`1` similarly to what `ZEND_THREEWAY_COMPARE` does?

Reply:
It returns a mask.

Reply:
A mask of what?

Reply:
Those values added on the stub :

```c
#define GD_CMP_IMAGE		1
#define GD_CMP_NUM_COLORS	2
#define GD_CMP_COLOR		4
#define GD_CMP_SIZE_X		8
#define GD_CMP_SIZE_Y		16
#define GD_CMP_TRANSPARENT	32
#define GD_CMP_BACKGROUND	64
#define GD_CMP_INTERLACE	128
#define GD_CMP_TRUECOLOR	256
```

Reply:
So 0 => no difference or one or more of those values for each difference found.